swim/disseminator.go

```diff
@@ -88,7 +88,10 @@ func (d *disseminator) AdjustMaxPropagations() {
 
 // HasChanges reports whether disseminator has changes to disseminate.
 func (d *disseminator) HasChanges() bool {
-	return len(d.changes) > 0
+	d.Lock()
+	r := len(d.changes) > 0
```

Comment:
I believe it's quite idiomatic to have 1 letter variable names, especially in small scope functions like this.